pkg/svc/workflow.go

```diff
@@ -112,21 +116,51 @@ func (s *workflowsrvc) Assign(ctx context.Context, w *workflow.AssignPayload) (e
 	return nil
 }
 
+// ListAssignments lists workflow assignments
+func (s *workflowsrvc) ListAssignments(ctx context.Context, w *workflow.ListAssignmentsPayload) (res []*workflow.WorkflowAssignment, err error) {
+	listeners := map[string]*workflow.WorkflowListener{}
+	for wf, codesets := range s.store.GetAssignments(ctx, w.Name) {
+		var listener *workflow.WorkflowListener
+		if l, ok := listeners[wf]; ok {
+			listener = l
+		} else {
+			listener, err = s.backend.GetWorkflowListener(ctx, s.logger, wf)
+			if err != nil {
+				s.logger.Print(err)
+				return nil, err
+			}
+			listeners[wf] = listener
+		}
```

Comment:
This is exactly the interaction between the REST API service and the Tekton backend I was referring to earlier. You can avoid this by doing the mapping I suggested directly inside the workflow store. This code shouldn't need to do anything this elaborate, it should be really thin and act only as a gateway to the workflow store.

Reply:
Yes, this is part of the discussion about where to put the logic which is currently mixed between the store and workflow.go, probably because `workflow.go` is the only place where is possible to talk with both workflow backend and workflow store.

I think there are many options here, for example we can introduce a new concept (`controller`?) where we make the connection between workflow store/backend and make `workflow.go` operate over the store and backend through this `controller`.

Since we did not agree on it yet, I propose leaving it as is until we reach a decision.

Reply:
ok to leaving it as it is